Posts & Pages: Add "deleting..." state

kean opened the pull request:
- Move deletion and restoration to `PostCoordinator` 
- Remove the alert after tapping “Undo”: the post will simply appear where it was, which is enough
- Fix an issue with SiteSettings view controller displayed in the prompt of 403 not having a “Done” button

Notes:
- Permanent deletion is currently optimistically deletes the post (production behavior)
- Pages don't have a respective status label

## To test:

Follow the steps from the video:

https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/c4819eb7-3e31-4458-b57d-2cd78b7a2003



## Regression Notes
1. Potential unintended areas of impact: Posts 
2. What I did to test those areas of impact (or what existing automated tests I relied on): manual
3. What automated tests I added (or what prevented me from doing so): n/a

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

kean committed:
Move post deletion to PostCoordinator

kean committed:
Update post cell status when post is being deleted

kean committed:
Add support for post deletion status in search

kean committed:
Animate reloads

momo-ozawa reviewed (commented):
@kean Nice improvements!

### Restored post always shows up in drafts
This is the behavior on production as well. A post restored via the "Undo" action is always restored as a draft.

If we remove the alert, let's make sure to restore the posts in the original status / tab. Not sure if this is possible though - in which case we may have to keep the undo alert.

https://github.com/wordpress-mobile/WordPress-iOS/assets/6711616/bbb2c2f4-7672-4f05-b8bc-9e5b69edf1be

### Footer cell w activity indicator becomes visible when post is trashed

After trashing a post, the footer cell becomes visible, and doesn't go away even after the post has been trashed.

I don't think we need to show the footer cell when deleting a post - wdyt?

https://github.com/wordpress-mobile/WordPress-iOS/assets/6711616/9ad8f400-1ba2-4501-b9d0-ce445b75e871

kean committed:
Show latest posts in search

kean committed:
Merge pull request #21913 from wordpress-mobile/task/show-latest-posts-in-search

Posts & Pages: Show latest posts in search

kean committed:
Remove Undo button

kean commented:
I removed the "Undo" button as discussed in Slack p1698763051200289-slack-C04SFL0RP51 and added the footer bug, which I think is a regression introduced in one of the previous PRs, to https://github.com/wordpress-mobile/WordPress-iOS/issues/21900.

I also unintentionally merged https://github.com/wordpress-mobile/WordPress-iOS/pull/21913 in this PR, which was already approved.
